File: LeetCode/Java/0171. Excel Sheet Column Number.java
class Solution {
    public int titleToNumber(String columnTitle) {
        int n = columnTitle.length();
        int columnNumber = 0;
        int mapping;
        
        for (int i = 0; i < n; i++) {
            mapping = columnTitle.charAt(i) - 'A' + 1;
            columnNumber += mapping * (int) Math.pow(26, n - i - 1);
        }
        return columnNumber;
    } 
}

File: LeetCode/Python/0171. Excel Sheet Column Number.py
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        column_number = 0
        
        for i, char in enumerate(reversed(columnTitle)):
            mapping = ord(char) - ord('A') + 1
            column_number += pow(26, i) * mapping    
        return column_number
